app/models/report.rb

```diff
@@ -39,6 +46,28 @@ def self.clear_cache
     end
   end
 
+  def self.wrap_slow_query(timeout = 20000, &block)
+    begin
+      ActiveRecord::Base.connection.transaction do
+        # Set a statement timeout so we can't tie up the server
+        DB.exec "SET LOCAL statement_timeout = #{timeout}"
```

Comment:
I'm not 100% sure but I think `DB.exec` might not be run in the context of this transaction because it bypasses AR.

Reply:
It seems to work and this is also what we do in data explorer plugin: https://github.com/discourse/discourse-data-explorer/blob/master/plugin.rb#L90

Reply:
yes it will always run within the transaction AR makes.